### DISCUSSION TITLE
[poisson-disk-sampling] issue with argument types

### DISCUSSION THREAD

**Discussion by cdaein** [DefinitelyTyped/DefinitelyTyped#63176] [Issues with a @types package]
There are 2 issues I want to address...

1.
According to the original `poisson-disk-sampling` [source code](https://github.com/kchapelier/poisson-disk-sampling/blob/9536e50c682c8e1d5a5add44a729b7414a276c61/src/implementations/variable-density.js#L36
),
the constructor's `distanceFunction` argument takes a `point` argument that is `Array` of any length. So, the proper type will be `number[]`. The current type definition has a spread operator like below, which indicates this function may take multiple arguments. 

from [the current definition](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/0c4376c0b8386acae927499033312cbb7cf63c29/types/poisson-disk-sampling/index.d.ts#L19):
```
distanceFunction?: ((...params: number[]) => number) | null;
```

But, looking at the source implementation, it will take only one argument of `point` that can be of any length. So I think it should be changed to,

```
distanceFunction?: ((params: number[]) => number) | null;
```

2.
Another constructor argument `bias` is defined as `float` in the [source code](https://github.com/kchapelier/poisson-disk-sampling/blob/9536e50c682c8e1d5a5add44a729b7414a276c61/src/implementations/variable-density.js#L32).

So, the type can just be `number`
```
bias?: number;
```

@Aliyss - would you be able to review this as you made the original PR? I've only started to work with this library, so apologies if there's something I've mistaken. Thank you!

[a comment by an automated account is not shown]

**Comment by Semigradsky** [comment]
I have created the PR: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/63206

**Reply by aliyss** [reply]
Thanks a lot... Was quite busy this week. Appreciate the help!

**Reply by Semigradsky** [reply]
Could you review please?

**Comment by aliyss** [comment]
Should now be merged.
